### ISSUE TITLE
what is something better should be coming really soon

### ISSUE THREAD

**Issue by amorenew** [touchlab/knarch.threads#1]
Hi
What do you mean by something better should be coming really soon?
Thanks

**Comment by kpgalligan** [comment]
Hey! Sorry I missed this. In the case of the "threads" packages, they really only provide a wrapper for AtomicReference for native, and a ThreadLocal. The LiveData package provides a somewhat stripped down version of Google's LiveData.

The atomic stuff will be available in things like atomic.fu (I actually just published an implementation for native, so in that case, this is ["something better"](https://github.com/touchlab/kotlinx.atomicfu/blob/master/atomicfu-native/src/main/kotlin/kotlinx/atomicfu/AtomicFU.kt). Thread local needs some thinking, to be honest. The implementation works, but it's not ideal.

For LiveData, multithreaded coroutines for native should be coming at some point, although the ["couple weeks"](https://github.com/Kotlin/kotlinx.coroutines/issues/462) estimate hasn't panned out. However, I'm sure it's coming during or after Kotlinconf, and I have it on good authority that Jetbrains is going to make a reactive library for Kotlin.

If there's interest, keeping LiveData as an active project might make sense. The google version isn't a particularly large codebase (which is why I wrote this instead of trying to write something like Rx).
